### DISCUSSION TITLE
Commas not allowed in registry passwords

### DISCUSSION THREAD

**Discussion by btoews** [aquasecurity/trivy#7043] [Bugs]
### Description

Specifying a password that includes a comma results in an error about the number of specified usernames/passwords not matching.

### Desired Behavior

Authenticate to a registry with a password including a comma.

### Actual Behavior

An error message.

### Reproduction Steps

```bash
`TRIVY_USERNAME="foo" TRIVY_PASSWORD="bar,baz" trivy image whatever`
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
`
$ TRIVY_USERNAME="foo" TRIVY_PASSWORD="bar,baz" trivy -d image whatever
2024-06-27T07:13:04-06:00	FATAL	Fatal error	flag error: registry flag error: the length of usernames and passwords must match
`
```

### Operating System

macOS

### Version

```bash
main branch
```

### Checklist

- [ ] Run `trivy clean --all`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
In your case, we'd recommend using `docker login` or other tools to get logged in.
